modules/apps/headless/headless-portal-instances/headless-portal-instances-impl/src/main/java/com/liferay/headless/portal/instances/internal/resource/v1_0/PortalInstanceResourceImpl.java

```diff
@@ -27,4 +41,155 @@
 	scope = ServiceScope.PROTOTYPE, service = PortalInstanceResource.class
 )
 public class PortalInstanceResourceImpl extends BasePortalInstanceResourceImpl {
+
+	@Override
+	public void deletePortalInstance(String webId) throws Exception {
+		Company company = _companyLocalService.getCompanyByWebId(webId);
+
+		if (company.isActive()) {
+			throw new ValidationException(
+				"Deactivate the portal instance before deleting it");
+		}
+
+		if (_portalInstancesLocalService.getDefaultCompanyId() ==
+				company.getCompanyId()) {
+
+			throw new ValidationException(
+				"Select other default portal instance before deleting it");
+		}
+
+		_companyLocalService.deleteCompany(company.getCompanyId());
+	}
+
+	@Override
+	public PortalInstance getPortalInstance(String webId) throws Exception {
+		return _getPortalInstance(
+			_companyLocalService.getCompanyByWebId(webId));
+	}
+
+	@Override
+	public Page<PortalInstance> getPortalInstancesPage(String skipDefault)
+		throws Exception {
+
+		List<Company> companies = _companyLocalService.getCompanies(false);
+
+		boolean skipDefaultCompany = GetterUtil.get(skipDefault, false);
+
+		List<PortalInstance> portalInstances = new ArrayList<>();
+
+		for (Company company : companies) {
+			if (skipDefaultCompany &&
+				(_portalInstancesLocalService.getDefaultCompanyId() ==
+					company.getCompanyId())) {
+
+				continue;
+			}
+
+			portalInstances.add(_getPortalInstance(company));
+		}
+
+		return Page.of(portalInstances);
```

Comment:
Brian likes using transformArray to simplify this but because you are filtering before transforming the results I think it can be Ok